statsmodels/tsa/base/tsa_model.py

```diff
@@ -49,6 +51,8 @@ def _init_dates(self, dates, freq):
             if (not datetools._is_datetime_index(dates) and
                     isinstance(self.data, data.PandasData)):
                 try:
+                    if is_numeric_dtype(dates):
+                        raise ValueError
```

Comment:
This change was needed to ensure that indices like `[1,2,3...]` are not converted to dates, which `to_datetime` is happy to convert to (nano?) milliseconds past 1-1-1970.